File: Julia/day1.jl
f = open("day1_input.txt")

lines = readlines(f)

sumdiff = 0

list1 = []
list2 = []

for line in lines
    sline = split(line)

    push!(list1, parse(Int, sline[1]))
    push!(list2, parse(Int, sline[2]))
end

sort!(list1)
sort!(list2)

for i in 1:length(list1)
    global sumdiff += abs(list1[i] - list2[i])
end

simscore = 0

for l1element in list1
    for l2element in list2
        if l1element == l2element
            global simscore += l2element
        end
    end
end

println(sumdiff)
println(simscore)

File: Ruby/day1.rb
f = File.open('day1_input.txt')

lines = f.readlines()

sumdiff = 0

list1 = []
list2 = []

lines.each do |line|
  split_line = line.split(" ")

  list1.push(split_line[0].to_i)
  list2.push(split_line[1].to_i)
end

list1.sort!
list2.sort!

for i in 0..list1.length-1
  sumdiff += (list1[i] - list2[i]).abs
end

simscore = 0

list1.each do |l1element|
  list2.each do |l2element|
    if l1element == l2element
      simscore += l2element
    end
  end
end

puts sumdiff
puts simscore
